### PR TITLE
test: smoke tests with testscript

### PR DESCRIPTION
**Pull request by caarlos0**
started to add somes tests using `testscript`.

For now, a basic user management test case is added... still trying to figure out how to make coverage et al work, but so far, this looks great tbh

### CODE REVIEW

File: server/backend/sqlite/sqlite.go [hunk 1]
```diff
@@ -603,6 +603,9 @@ func (d *SqliteBackend) RemoveCollaborator(repo string, username string) error {
 }
 
 func (d *SqliteBackend) initRepo(repo string) error {
+	if os.Getenv("SOFT_SERVE_TEST_NO_HOOKS") == "1" {
```
[the remark this answers is not in the document]
Comment (aymanbagabas) [reply server/backend/sqlite/sqlite.go:606]: hmm this is hacky we should at least add a comment and todo item :/ another option would be to look for `test.v` arg https://stackoverflow.com/a/36666114/10913628

[the remark this answers is not in the document]
Comment (aymanbagabas) [reply server/backend/sqlite/sqlite.go:606]: probably better to move this into `GenerateHooks`

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -3,26 +3,26 @@ name: build
 on: [push, pull_request]
 
 jobs:
-  build:
-    uses: charmbracelet/meta/.github/workflows/build.yml@crlf
```
Comment (caarlos0) [inline .github/workflows/build.yml]: good one, lets see if everything still works

File: .editorconfig [hunk 3]
```diff
@@ -1,6 +1,6 @@
 root = true
 
-[*]
```
Comment (caarlos0) [inline .editorconfig]: oops, good call!

should probably move charset et all to [*] still though, doing it rn

**Review by aymanbagabas — APPROVED** [review]

**Comment by caarlos0** [comment]
thanks for the help here @aymanbagabas 💜

File: server/utils/utils.go [hunk 4]
```diff
@@ -2,15 +2,17 @@ package utils
...
 	"strings"
 	"unicode"
 )
 
 // SanitizeRepo returns a sanitized version of the given repository name.
 func SanitizeRepo(repo string) string {
 	repo = strings.TrimPrefix(repo, "/")
-	repo = filepath.Clean(repo)
+	// We're using path instead of filepath here because this is not OS dependent
+	// looking at you Windows
```
Comment (bashbunni) [inline server/utils/utils.go:14]: LOL

**Pull request merged**